modules/subscriptionService/src/main/scala/routes/Routes.scala

```diff
@@ -18,14 +19,54 @@ object Routes {
       jsonEncoderOf[F, PostSubscriptions]
     implicit val decoder: EntityDecoder[F, PostSubscriptions] =
       jsonOf[F, PostSubscriptions]
+    implicit val slackCommandRequestBodyEncoder: EntityEncoder[F, SlackCommandRequestBody] =
+      jsonEncoderOf[F, SlackCommandRequestBody]
+    implicit val slackCommandRequestBodyDecoder: EntityDecoder[F, SlackCommandRequestBody] =
+      jsonOf[F, SlackCommandRequestBody]
 
     HttpRoutes.of[F] {
-      case GET -> Root                => Ok("subscription service")
-      case GET -> subscription / user => Ok(MockedResponse.mockedGetUserSubscriptionResponse)
-      case req @ POST -> subscription / user =>
-        req.decode[PostSubscriptions] { message: PostSubscriptions =>
-          Created(message)
-        }
+      case GET -> Root => Ok("subscription service")
+      case GET -> Root / "subscription" / user =>
+        Ok(MockedResponse.mockedGetUserSubscriptionResponse)
+      case req @ POST -> Root / "subscription" / user =>
+        handleRequest[F, PostSubscriptions](req, "application/json")
+      case req @ DELETE -> Root / "subscription" / user =>
+        handleRequest[F, PostSubscriptions](req, "application/json")
+      case req @ POST -> Root / "subscription" / "slack" / command =>
+        req
+          .withHeaders(
+            Headers.of(
+              Header("Content-Type", "application/x-www-form-urlencoded")
+            )
+          )
+          .attemptAs[SlackCommandRequestBody]
+          .foldF(
+            error => BadRequest(s"An error occurred ${error}"),
+            _ => Accepted()
+          )
     }
   }
+
+  private def handleRequest[F[_]: Sync, A <: Subscription](
```

Comment:
I like this approach, but I would:
* Move this method inside of the `subscription` method (so we can reuse the `dsl`)
* Remove the `contentType`, we don't need to specify it to decode the request

Reply:
@fedefernandez please I have resolved all the comments on the PR, thanks.